services/web-server/src/login/strategies/github.js

```diff
@@ -82,8 +123,15 @@ module.exports = class Github {
           clientID: strategyCfg.clientId,
           clientSecret: strategyCfg.clientSecret,
           callbackURL: `${cfg.app.publicUrl}${callback}`,
+          scope: 'repo',
         },
         async (accessToken, refreshToken, profile, next) => {
+          this.clients.set(Number(profile.id), new GithubClient({ accessToken }));
+          await this.GithubAccessToken.create({
+            userId: profile.id,
+            accessToken,
+            expires: taskcluster.fromNow('10 minutes'),
```

Comment:
10 minutes doesn't seem a very long time to be logged in.  How long is the access token from GH valid?

Reply:
The token from GH never expires. You're right, 10 minutes is not a very long time. I didn't think when setting the expiration time that this will be used by the scanner process. I could change it to something like 1 week.

Reply:
Oh, cool, never is good!  Then we could probably keep them in this table forever, too?

So I guess the "expiration" is when a user revokes access, at which point the access token becomes invalid and the scanner should then disable all of their clients (rather than crash).  Which makes sense: revoke access, lose credentials.

I don't think there's a reason to try to purge revoked tokens from the table -- there is only one row per user, so the table will have a fairly stable size.

Reply:
Correct. Also, forever sounds good to me :)

> I don't really understand how this will work. Can you outline it?

Let me know if you still don't understand how this will work. I can try to explain.